dvc/tree/s3.py

```diff
@@ -54,18 +54,32 @@ def __init__(self, repo, config):
 
         self._append_aws_grants_to_extra_args(config)
 
+        self.key_id = config.get("s3_key_id")
+        self.key_secret = config.get("s3_key_secret")
+
```

Comment:
I don't think this is going to work, as you didn't enable these fields in `dvc/config.py`.